src/main.cpp

```diff
@@ -4867,6 +4873,157 @@ void static ProcessGetData(CNode* pfrom)
     }
 }
 
+bool checkVerificationQueue(){
+    return verificationQueue.isEmpty();
+}
+
+void txVerificationThread(){
+    int i = 0;
+    while(!ShutdownRequested())
+    {
+        if (!checkVerificationQueue() && i < 100)
```

Comment:
In order to have the LOCK protecting the whole access to the queue, wouldnt it be better this:

        if (i < txVerificationLimit)
        {
            LOCK(cs_verificationQueue);
            if (!(verificationQueue.isEmpty()) )
            {
                LogPrintf("%s():%d - CONS: deque is empty[%d], size[%d]\n", __func__, __LINE__,
                    verificationQueue.dequeTX.empty(), verificationQueue.dequeTX.size() );
                verificationQueue.verifyOne();
                LogPrintf("%s():%d - CONS: deque is empty[%d], size[%d]\n", __func__, __LINE__,
                    verificationQueue.dequeTX.empty(), verificationQueue.dequeTX.size() );
                i++;
            }
        }
        else
        {
            i = 0;
            std::this_thread::sleep_for (std::chrono::seconds(1));
        }

Reply:
Done but in little bit different way

https://github.com/andreysobol/zen/commit/2133ee092b4b0dbac7338c0a40b5be2593ddbbe6

```
    int i = 0;
    bool emptyQueue = false;
    while(!ShutdownRequested())
    {
        if ((!emptyQueue) && i < txVerificationLimit)
        {
            LOCK(cs_verificationQueue);
            if(!(verificationQueue.isEmpty())){
                verificationQueue.verifyOne();
                i++;
            }
            else{
                emptyQueue = true;
            }
        }
        else
        {
            i = 0;
            emptyQueue = false;
            std::this_thread::sleep_for (std::chrono::seconds(1));
        }
    }
```